src/lib/esnext.intl.d.ts

```diff
@@ -1,3 +1,168 @@
 declare namespace Intl {
-    // Empty
+    type DurationTimeFormatLocaleMatcher = "lookup" | "best fit";
+    /**
+     * Value of the `unit` property in duration objects
+     *
+     * [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DurationFormat/format#duration).
+     */
+    type DurationTimeFormatUnit = "years"|"months"|"weeks"|"days"|"hours"|"minutes"|"seconds"|"milliseconds"|"microseconds"|"nanoseconds"
+    
+    /**
+     * The style of the formatted duration.
+     *
+     * [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DurationFormat/DurationFormat#style).
+     */
+    type DurationFormatStyle = "long" | "short" | "narrow" | "digital"
+
+    
+    type DurationFormatUnitSingular =
+    | "year"
+    | "quarter"
+    | "month"
+    | "week"
+    | "day"
+    | "hour"
+    | "minute"
+    | "second";
+
+    /**
+     * An object representing the relative time format in parts
+     * that can be used for custom locale-aware formatting.
+     *
+     * [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DurationFormat/formatToParts#examples).
+     */
+        type DurationFormatPart =
+        | {
+            type: "literal";
+            value: string;
+        }
+        | {
+            type: Exclude<NumberFormatPartTypes, "literal">;
+            value: string;
+            unit: DurationFormatUnitSingular;
+        };
+
+
+
+    type DurationFormatOption = "long"|"short"|"narrow"|"numeric"|"2-digit"
+
+
+    type DurationFormatDisplayOption = "always"|"auto";    
+
+    /**
+     * Number of how many fractional second digits to display in the output.
+     *
+     * [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DurationFormat/DurationFormat#fractionaldigits).
+     */
+    type fractionalDigitsOption = 0|1|2|3|4|5|6|7|8|9

```

Comment:
Out of curiosity, is it intentional that this type starts with lowercase? It looks a bit out of place as all the others are in PascalCase.